### PR TITLE
ESP32/wifi-echo: Generate QR code display link for devices without display

### PR DESCRIPTION
**Pull request by dhrishi**
#### Problem
 Need a mechanism to display QR code for devices that do not have a physical display

 #### Summary of Changes
  i. URL encode the QR code text and pass as data to the QR code display page.
  ii. We can later move this page to the CHIP repository.

### CODE REVIEW

File: examples/wifi-echo/server/esp32/main/wifi-echo.cpp [hunk 1]
```diff
@@ -54,6 +55,8 @@ using namespace ::chip::DeviceLayer;
 
 extern void startServer();
 
+#define QRCODE_BASE_URL "https://dhrishi.github.io/connectedhomeip/qrcode.html"
```
[the remark this answers is not in the document]
Comment (dhrishi) [reply examples/wifi-echo/server/esp32/main/wifi-echo.cpp:58]: @andy31415 Yes, it's possible, as it is an independent piece. That's why, I have already mentioned in the PR description that we can move the page to the CHIP repo in a follow-on PR.

[the remark this answers is not in the document]
Comment (rwalker-apple) [reply examples/wifi-echo/server/esp32/main/wifi-echo.cpp:58]: please file an issue and annotate here with a pointer to the issue

File: examples/wifi-echo/server/esp32/main/wifi-echo.cpp [hunk 2]
```diff
@@ -421,6 +424,48 @@ std::string createSetupPayload()
     return result;
 };
 
+bool inRfc3986UnreservedChar(const char c)
+{
+    char unreserved[] = { '-', '.', '_', '~' };
```
Comment (rwalker-apple) [inline examples/wifi-echo/server/esp32/main/wifi-echo.cpp:429]: `static const char *` ?

**Review by rwalker-apple — APPROVED** [review]
other than nits, lgtm

**Review by kedars — APPROVED** [review]

File: examples/wifi-echo/server/esp32/main/wifi-echo.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (dhrishi) [reply examples/wifi-echo/server/esp32/main/wifi-echo.cpp:58]: I have created a new issue here: #3266

File: examples/wifi-echo/server/esp32/main/wifi-echo.cpp (hunk 2 above)
Comment (dhrishi) [reply examples/wifi-echo/server/esp32/main/wifi-echo.cpp]: Used `static const char []` for better readability

**Review by BroderickCarlin — APPROVED** [review]

File: examples/wifi-echo/server/esp32/main/wifi-echo.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (dhrishi) [reply examples/wifi-echo/server/esp32/main/wifi-echo.cpp:58]: Is there anything else that needs to be done here?

**Comment by mspang** [comment]
Reminder to use the "re-request review button"

**Review by mspang — APPROVED** [review]

**Pull request merged**